src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2443,6 +2443,18 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                     GenerateImplicitConversionError(diagnostics, expression.Syntax, conversion, expression, targetType);
                 }
 
+                if (expression.Kind == BoundKind.TupleLiteral)
+                {
+                    // A BoundTupleLiteral must always be upgraded to a BoundConvertedTupleLiteral during conversion
+                    var sourceTuple = (BoundTupleLiteral)expression;
+                    expression = new BoundConvertedTupleLiteral(
+                        sourceTuple.Syntax,
+                        sourceTuple.Type,
+                        sourceTuple.Arguments,
+                        sourceTuple.Type ?? targetType,
```

Comment:
Yes, there is no right answer here since the conversion is not possible. It feels better to keep natural type if such exists, otherwise returning error type seems appropriate, if it is ok with other expectations.